File: src/main/java/com/extendedae_plus/mixin/impl/instance/ImplClientOnly.java
package com.extendedae_plus.mixin.impl.instance;

import com.extendedae_plus.mixin.impl.HelperClientOnly;
import net.minecraft.client.gui.screens.Screen;

public class ImplClientOnly implements HelperClientOnly {
    @Override
    public boolean hasControlDown() {
        return Screen.hasControlDown();
    }

    @Override
    public boolean hasShiftDown() {
        return Screen.hasShiftDown();
    }
}

File: src/main/kotlin/com/extendedae_plus/client/impl/ImplClientOnly.kt
package com.extendedae_plus.client.impl

import com.extendedae_plus.util.UtilClient
import net.minecraft.client.gui.screens.Screen

object ImplClientOnly : UtilClient {
    override fun _shift() = Screen.hasShiftDown()

    override fun _ctrl() = Screen.hasControlDown()

    override fun _alt() = Screen.hasAltDown()
}